### PR TITLE
[DD-962] Regexlookup

### PR DESCRIPTION
**Pull request by janpolowinski**

### CODE REVIEW

**Comment by boferri** [comment]
@janpolowinski please add the unit test from the d:swarm backend ( converter/src/test/java/org/dswarm/converter/morph/RegexLookupTest.java) here.

**Comment by janpolowinski** [comment]
Ok, I can move the test here. At first glance I wasn't sure if we want this test here, since metafacture uses some xml-based tests (or at least some mixture of XML ans Java).

**Comment by boferri** [comment]
I think it doesn't really matter, whether we make use of their test framework or simple write a plain unit test. At the end I think that it is more important to have the test where the functionality is located ;)

**Comment by janpolowinski** [comment]
Jenkins has a file here, as the workspace view shows: 

http://194.95.145.11/job/Github%20PR%20Metafacture/org.culturegraph$metafacture-core/ws/target/test-classes/data/regexlookup.csv

and searches here:

/var/lib/jenkins/jobs/Github%20PR%20Metafacture/workspace/target/test-classes/data/regexlookup.csv

Looks like the correct folder for me. Did someone have similar problems before?

**Comment by boferri** [comment]
Maybe you can try to load the file with Resources class from Guava (Resources#getResource([relative file name]) -> URL + Resources#toString([url], [charset])). So you first need to include Guava as test dependency in the project. Usually the ${basedir} isn't required at the resource directory + testResource directory configuration in the project pom.

File: src/test/resources/morph/regexlookup.morph.xml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+   		<data name="newName" source="attributes">
+<!--    			<whitelist map="normalMap"/> -->
+<!--    			<lookup in="normalMap"/> -->
+<!--    		<replace  pattern="^J(.).*"  with="X$1"/> -->
+<!--    		<replace  pattern="[ony]"  with="X"/> -->
+	   		<regexlookup map="regexMap">
+<!--             	 <entry name="^J(.).*" value="1"/> -->
+<!--                 <entry name="^Ja(.).*" value="2"/> -->
+<!--                 <entry name="^J.{0,20}" value="3"/> -->
+			</regexlookup>
```
Comment (fniederl) [inline src/test/resources/morph/regexlookup.morph.xml:9]: Perhaps the comments could be deleted?

Comment (janpolowinski) [inline src/test/resources/morph/regexlookup.morph.xml:9]: ok, good idea since the data is also cleaned up ;)

**Pull request merged**